tests/Reminder/Reminder.test.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * @jest-environment jsdom
+ */
+jest.mock('obsidian');
+import moment from 'moment';
+import { Reminder, ReminderType, parseMoment } from '../../src/Reminders/Reminder';
+
+window.moment = moment;
+
+describe('Reminder ', () => {
+    it('testing parsing Moment() datetime', () => {
+        const reminder = moment('2023-04-30 11:44 am');
+
+        expect(parseMoment(reminder)).toStrictEqual(new Reminder(moment('2023-04-30 11:44 am'), ReminderType.DateTime));
+    });
+
+    it('testing parsing Moment() date', () => {
+        const reminder = moment('2023-04-30');
+
+        expect(parseMoment(reminder)).toStrictEqual(new Reminder(moment('2023-04-30'), ReminderType.Date));
+    });
+});
```

Comment:
I wanted to understand how `parseDateTime()` worked in the two time formats, so I wrote some tests and thought you might like to add them to your code, to save you from duplicating effort....

I put this in `Reminder.test.ts:


```ts
describe('should parse reminder dates and times', () => {
    afterEach(function () {
        resetSettings();
    });

    function setDateTimeFormat(dateTimeFormat: string) {
        const settings = getSettings().reminderSettings;
        settings.dateTimeFormat = dateTimeFormat;
        updateSettings({ reminderSettings: settings });
    }

    function checkParsedDateTime(input: string, output: string) {
        const dateTime = parseDateTime(input);
        expect(dateTime).not.toBeNull();
        expect(dateTime!.toString()).toEqual(output);
    }

    it('should parse valid dates and times - in 12-hour format', () => {
        setDateTimeFormat(TIME_FORMATS.twelveHour);

        checkParsedDateTime('2023-01-15', '2023-01-15');
        checkParsedDateTime('2024-01-15 13:45', '2024-01-15 1:45 pm'); // 12-hour format reads 24-hour OK
        checkParsedDateTime('2023-01-15 1:45 am', '2023-01-15 1:45 am');
        checkParsedDateTime('12/13/2019', 'Invalid date');
    });

    it('should parse valid dates and times - in 24-hour format', () => {
        setDateTimeFormat(TIME_FORMATS.twentyFourHour);

        checkParsedDateTime('2023-01-15', '2023-01-15');
        checkParsedDateTime('2023-01-15 13:45', '2023-01-15 13:45');
        checkParsedDateTime('2023-01-15 1:45 pm', '2023-01-15 01:45'); // the pm is ignored
    });
});
```

Reply:
I actually think the results of this are really interesting.

When 12-hour format is selected, it does actually read 24-hour times too.

So it's more forgiving than I expected.

This will be important to mention in the user documentation please.

Reply:
It would be good to see similar kinds of tests in the Task-parsing code - to confirm the behaviour at that level, where regular expressions have an impact on the behaviour too.

Reply:
Thanks for this